backend/plugin/parser/mysql/query_test.go

```diff
@@ -10,49 +10,60 @@ import (
 
 func TestValidateSQLForEditor(t *testing.T) {
 	tests := []struct {
-		statement string
-		validate  bool
-		err       bool
+		statement   string
+		valid       bool
+		gotAllQuery bool
+		err         bool
 	}{
 		{
-			statement: "SHOW CREATE TABLE bytebase;",
-			validate:  true,
+			statement:   "SHOW CREATE TABLE bytebase;",
+			valid:       true,
+			gotAllQuery: true,
 		},
 		{
-			statement: "DESC bytebase;",
-			validate:  true,
+			statement:   "DESC bytebase;",
+			valid:       true,
+			gotAllQuery: true,
 		},
 		{
-			statement: "SELECT * FROM t1 WHERE c1 = 1; SELECT * FROM t2;",
-			validate:  true,
+			statement:   "SELECT * FROM t1 WHERE c1 = 1; SELECT * FROM t2;",
+			valid:       true,
+			gotAllQuery: true,
 		},
 		{
-			statement: "CREATE TABLE t1 (c1 INT);",
-			validate:  false,
+			statement:   "CREATE TABLE t1 (c1 INT);",
+			valid:       false,
+			gotAllQuery: false,
 		},
 		{
-			statement: "UPDATE t1 SET c1 = 1;",
-			validate:  false,
+			statement:   "UPDATE t1 SET c1 = 1;",
+			valid:       false,
+			gotAllQuery: false,
 		},
 		{
-			statement: "EXPLAIN SELECT * FROM t1;",
-			validate:  true,
+			statement:   "EXPLAIN SELECT * FROM t1;",
+			valid:       true,
+			gotAllQuery: true,
 		},
 		{
-			statement: "EXPLAIN FORMAT=JSON DELETE FROM t1;",
-			validate:  false,
+			statement:   "EXPLAIN FORMAT=JSON DELETE FROM t1;",
+			valid:       false,
+			gotAllQuery: false,
 		},
 		{
-			statement: `select* from t`,
-			validate:  true,
+			statement:   `select* from t`,
+			valid:       true,
+			gotAllQuery: true,
 		},
 		{
-			statement: `explain select * from t;`,
-			validate:  true,
+			statement:   `explain select * from t;`,
+			valid:       true,
+			gotAllQuery: true,
 		},
 		{
-			statement: `explain    analyze select * from t`,
-			validate:  true,
+			statement:   `explain    analyze select * from t`,
+			valid:       true,
+			gotAllQuery: true,
```

Comment:
`explain analyze` is not the read-only statement

Reply:
It's same to TIDB

Reply:
I will do it in a follow-up PR.